shell/platform/android/io/flutter/embedding/engine/systemchannels/PlatformChannel.java

```diff
@@ -581,6 +503,147 @@ public SystemChromeStyle(
     }
   }
 
+  /**
+   * A handler of incoming platform channel method calls received from Flutter.
+   * It first determines the platform's API to be called. If it exists, it then
+   * decodes incoming arguments, if needed, into a format that is necessary for the API.
+   */
+  @VisibleForTesting
+  protected class PlatformMethodCallHandler implements MethodChannel.MethodCallHandler {
```

Comment:
I'd like for us to avoid non-static inner classes. I think in this case I'd like for us to revert the movement of this handler definition back to where it was. But in future cases where we do define inner classes, they should be static to avoid the possibility of accidentally leaking references to the outer class.

Reply:
I had a discussion with @mklim and we felt that it was difficult to read an inline class definition within another class. I'll revert this in a subsequent PR